### DISCUSSION TITLE
could you have example how to use fitGAM with 20 cores

### DISCUSSION THREAD

**Discussion by FADHLyemen** [statOmics/tradeSeq#169] [Q&A] [answered]
I need to use 20 cores for fitGAM function:
I did 
`options(MulticoreParam=quote(MulticoreParam(workers=20)))`

then run fitGAM

```
sce_PT <- fitGAM(sce_PT3, conditions = factor(colData(sce_PT3)$t2dctrl),
                 genes = rownames(sce_PT3)[rownames(sce_PT3) %in% hvg_genes ],BPPARAM = BiocParallel::bpparam())

but still, fitGAM used all the cores. any suggestions?


```

**Comment by HectorRDB** [comment, marked as answer]
Hi @FADHLyemen 
Please refer to the [vignette where we discuss parallelization](https://statomics.github.io/tradeSeq/articles/fitGAM.html#parallel-computing), it should help you sort this out.
Best

**Comment by FADHLyemen** [comment]
I did 
```
BPPARAM <- BiocParallel::bpparam()
BPPARAM 
BPPARAM$workers <- 10 # use 10 cores
sce_subset <- fitGAM(counts=sce_subset@assays@data$counts,pseudotime =t,cellWeights=w, 
                     conditions = factor(colData(sce_subset)$sampletype),
                 genes = rownames(sce_subset)[rownames(sce_subset) %in% hvg_genes_Tcells ],
               parallel=TRUE, BPPARAM = BPPARAM
             )
```

But still, fitGAM used all cores.

**Comment by HectorRDB** [comment]
Just to check, how do you verify how many cores are used? Depending on your settings, R might do multi-threading on top of parallelization, which would make more than 10 R processes appear, even though only 10 cores are used for example.
